CONTRIBUTING.md

```diff
@@ -30,6 +30,22 @@ works.
 
 This document will guide you through the contribution process.
 
+### Rules of thumb
+
+1. __Provide motivation for the change__  
+   Why will this change make the code better; does it fix a but, is it a new
+   feature, etc.
+2. __Don't cause _unnecessary_ code churn__  
+   As part of the projects strategy we maintain several lines of code, code 
+   churn might hinder back-porting changes to other branches. Also when you 
+   change a line, your name will come up in `git blame` and might hide the
+   previous writer of the code.
```

Comment:
"churn" is also quite subjective. At the code and learn events, we *intentionally* encourage participants to focus on very small changes that end up leading to quite a bit of churn individually. We have also become much better at very quickly reviewing and landing smaller changes.